File: be-ojt2026/Services/Inventory/Inventory.Domain/Entities/Customer.cs
using System;
using System.Collections.Generic;

namespace Inventory.Domain.Entities;

public partial class Customer
{
    public int Id { get; set; }

    public string CustomerName { get; set; } = null!;

    public string? Phone { get; set; }

    public string? Email { get; set; }

    public string? Address { get; set; }

    public bool IsActive { get; set; }

    public DateTime? CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }

    public virtual ICollection<SaleOrder> SaleOrders { get; set; } = new List<SaleOrder>();
}

File: fe-ojt2026/src/types/customer.ts
// Types for Customer Management API

export interface Customer {
  Id: number;
  CustomerName: string;
  Email: string;
  Phone: string;
  Address: string | null;
  IsActive: boolean;
  CreatedAt: string;
  UpdatedAt: string | null;
}

export interface CustomerResponse {
  Items: Customer[];
  Page: number;
  PageSize: number;
  TotalItems: number;
  TotalPages: number;
  HasNext: boolean;
  HasPrevious: boolean;
}

export interface CreateCustomerRequest {
  email: string;
  name: string;
  phone: string;
  address?: string;
  isActive?: boolean;
}

export interface UpdateCustomerRequest {
  email: string;
  name: string;
  phone: string;
  address?: string;
  isActive?: boolean;
}

export interface GetCustomerParams {
  page?: number;
  pageSize?: number;
  isActive?: boolean;
  search?: string;
}
